### ISSUE TITLE
Reduce the size of the binary

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/cli#2109]
As of today, `tkn` is a very **big** binary.

```
$ du bin/tkn
147M	bin/tkn
$ strip bin/tkn
$ du bin/tkn
118M	bin/tkn
```

We should try to find some ways to reduce the size of it somehow 👼🏼 🙏🏼. Some ideas:

- Track tektoncd/pipeline deps and understand where we can limit pulling k8s.io ones
- Have our own "struct" representing the types ? (generated)

**Comment by chmouel** [comment]
cleanup deps will be the first thing that come in mind, we have too many dependencies that i am not sure it's worth to have

**Comment by aThorp96** [comment]
I don't think this is necessarily the order of magnitude we'd be looking for, but building without DWARF debugging symbols (`-ldflags "-w"`) by about 14%.

**Comment by chmouel** [comment]
we def do already in released version

https://github.com/tektoncd/cli/blob/main/.goreleaser.yml#L19

**Comment by chmouel** [comment]
fyi the size in 2025 🙃 


```
% rm -f bin/tkn && make bin/tkn
% du bin/tkn
207M	bin/tkn
% rm -f bin/tkn && make bin/tkn FLAGS="-s -w"
% du bin/tkn
147M	bin/tkn
```

in chatgpt words it's a 40% increase 

![Image](https://github.com/user-attachments/assets/f2853158-6bdf-4b36-9179-2d2d39ec8640)

**Comment by aThorp96** [comment]
I took a look at the dependencies used right now. At least around 19M of the indirect dependencies are related to storage backends(?) and signing related libs imported indirectly through `tekton/chains`. Particularly, `envoyproxy/go-control-plane` (7.69M), `aws-sdk-go` (both v1 and v2) (cum. 8.9M), and `mongo-driver` (2.5M). I don't fully grok yet how components like `tekton/chains` are used in the CLI, but depending on how they're used it may be possible to omit some of these larger indirect dependencies by cleaning up how the component's modules are structured. In Python there's Optional Dependencies and Rust has Features, I recall Go having strategies to accomplish similar things, but I'm still pretty new to Go dependency management so I don't recall how

![Image](https://github.com/user-attachments/assets/4631c6a7-91ec-4463-83d2-7ed07ba4033e)

**Comment by alliasgher** [comment]
Opened #2843 for the goreleaser `-s` change.
